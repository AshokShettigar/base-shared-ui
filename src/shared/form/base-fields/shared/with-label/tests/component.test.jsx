import { shallow } from 'enzyme';

import WithLabelField from '../component';

describe('WithLabelField component', () => {
    const requiredProps = {
        children: <span>children</span>
    };

    it('should render component with default props', () => {
        const wrapper = shallow(<WithLabelField {...requiredProps} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with id prop', () => {
        const wrapper = shallow(<WithLabelField {...requiredProps} id="id" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with description prop', () => {
        const wrapper = shallow(<WithLabelField {...requiredProps} description="description" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with descriptionClassName prop', () => {
        const wrapper = shallow(<WithLabelField {...requiredProps} descriptionClassName="descriptionClassName" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with label prop', () => {
        const wrapper = shallow(<WithLabelField {...requiredProps} label="label" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with className prop', () => {
        const wrapper = shallow(<WithLabelField {...requiredProps} className="className" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with labelClassName prop', () => {
        const wrapper = shallow(<WithLabelField {...requiredProps} labelClassName="labelClassName" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('shouldnt render component', () => {
        const wrapper = shallow(<WithLabelField {...requiredProps} rendered={false} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with required prop', () => {
        const wrapper = shallow(<WithLabelField {...requiredProps} required />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with hidden prop', () => {
        const wrapper = shallow(<WithLabelField {...requiredProps} hidden />);

        expect(wrapper).toMatchSnapshot();
    });
});
