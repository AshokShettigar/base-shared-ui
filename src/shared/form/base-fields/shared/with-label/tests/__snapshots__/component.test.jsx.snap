// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithLabelField component should render component with className prop 1`] = `
<WithLabelField
  className="className"
  hidden={false}
  required={false}
>
  <span>
    children
  </span>
</WithLabelField>
`;

exports[`WithLabelField component should render component with default props 1`] = `
<WithLabelField
  hidden={false}
  required={false}
>
  <span>
    children
  </span>
</WithLabelField>
`;

exports[`WithLabelField component should render component with description prop 1`] = `
<WithLabelField
  description="description"
  hidden={false}
  required={false}
>
  <span>
    children
  </span>
</WithLabelField>
`;

exports[`WithLabelField component should render component with descriptionClassName prop 1`] = `
<WithLabelField
  descriptionClassName="descriptionClassName"
  hidden={false}
  required={false}
>
  <span>
    children
  </span>
</WithLabelField>
`;

exports[`WithLabelField component should render component with hidden prop 1`] = `
<div
  hidden={true}
>
  <WithLabelField
    hidden={true}
    required={false}
  >
    <span>
      children
    </span>
  </WithLabelField>
</div>
`;

exports[`WithLabelField component should render component with id prop 1`] = `
<WithLabelField
  hidden={false}
  id="id"
  required={false}
>
  <span>
    children
  </span>
</WithLabelField>
`;

exports[`WithLabelField component should render component with label prop 1`] = `
<WithLabelField
  hidden={false}
  label="label"
  required={false}
>
  <span>
    children
  </span>
</WithLabelField>
`;

exports[`WithLabelField component should render component with labelClassName prop 1`] = `
<WithLabelField
  hidden={false}
  labelClassName="labelClassName"
  required={false}
>
  <span>
    children
  </span>
</WithLabelField>
`;

exports[`WithLabelField component should render component with required prop 1`] = `
<WithLabelField
  hidden={false}
  required={true}
>
  <span>
    children
  </span>
</WithLabelField>
`;

exports[`WithLabelField component shouldnt render component 1`] = `null`;
